neurostore/resources/data.py

```diff
@@ -168,72 +185,112 @@ def search(self):
         # record with most/all of the same details (e.g., DOI for studies)
-        data = parser.parse(self.schema, request)
-        record = self.__class__.update_or_create(data)
-        return self.schema().dump(record)
 
+        # Parse arguments using webargs
+        args = parser.parse(self._user_args, request, location="query")
+        source_id = args.get('source_id')
+        source = args['source'] or 'neurostore'
+        if source_id:    
+            data = self._load_from_source(source, source_id)
+        else:
+            data = parser.parse(self.__class__._schema, request)
 
-class DatasetsView(ObjectView):
-    _model = Dataset
+        record = self.__class__.update_or_create(data)
+        return self.__class__._schema().dump(record)
 
+@view_maker
+class DatasetView(ObjectView):
+    pass
 
-class StudiesView(ObjectView):
-    _model = Study
+
+@view_maker
+class StudyView(ObjectView):
     _nested = {
-        "analyses": "AnalysesView",
+        "analyses": "AnalysisView",
     }
 
 
-class AnalysesView(ObjectView):
-    _model = Analysis
+@view_maker
+class AnalysisView(ObjectView):
     _nested = {
-        "images": "ImagesView",
-        "points": "PointsView",
+        "images": "ImageView",
+        "points": "PointView",
     }
 
 
-class ConditionsView(ObjectView):
-    _model = Condition
-
+@view_maker
+class ConditionView(ObjectView):
+    pass
 
-class ImagesView(ObjectView):
-    _model = Image
 
+@view_maker
+class ImageView(ObjectView):
+    pass
 
-class PointsView(ObjectView):
-    _model = Point
+@view_maker
+class PointView(ObjectView):
     _nested = {
         "values": "PointValueView",
     }
 
-
+@view_maker
 class PointValueView(ObjectView):
-    _model = PointValue
+    pass
 
 
-class StudiesListView(ListView):
-    _model = Study
-    # _only = ('name', 'description', 'doi', '_type', '_id', 'created_at')
+@view_maker
+class StudyListView(ListView):
+    _nested = {
+        "analyses": "AnalysisView",
+    }
     _search_fields = ("name", "description")
 
+    @classmethod
+    def _load_from_source(cls, source, source_id):
+        if source == "neurostore":
+            return cls.load_from_neurostore(source_id)
+        elif source == "neurovault":
+            return cls.load_from_neurovault(source_id)
+        elif source == "pubmed":
+            return cls.load_from_pubmed(source_id)
+    
+    @classmethod
+    def load_from_neurostore(cls, source_id):
+        study = cls._model.query.filter_by(id=source_id).first_or_404()
+        data = cls._schema(copy=True).dump(study)
+        data['source'] = "neurostore"
+        data['source_id'] = source_id
+        data['source_updated_at'] = study.updated_at or study.created_at
+        return data
+
+        
+
+    @classmethod
+    def load_from_neurovault(cls, source_id):
+        pass
+
+    @classmethod
+    def load_from_pubmed(cls, source_id):
+        pass
+
```

Comment:
@tyarkoni, is this structure what you were thinking for cloning/copying from sources.